app/scripts/services/membership/roles.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  .factory('RolesService', function($q, DataService) {
+
+    var getClusterRoles = function() {
+      return DataService.get('clusterroles', null, {});
+    };
+
+    var getRoles = function(context) {
+      return DataService.get('roles', null, context);
+    };
+
+    // return set of clusterRoles & local roles
+    var getAllRoles = function(context) {
+      return $q.all([
+        getRoles(context),
+        getClusterRoles()
+      ]);
+    };
+
+    var listClusterRoles = function(cb, opts) {
+      DataService.list('clusterroles', {}, cb, opts);
+    };
+
+    var listRoles = function(context, cb, opts) {
+      DataService.list('roles', context, cb, opts);
+    };
+
+
+    // wraps the 2 API list calls in a promise for easier use, similar to above get()
+    var listAllRoles = function(context) {
+      var deferred = $q.defer();
+      var result = [];
+
+      var resolve = function(resp) {
+        result.push(resp.by('metadata.name'));
```

Comment:
this is really weird to me, you are returning an array with no guarantee of how things are being ordered.  Is there any reason not to return a single array of all the roles?